tests/test_model_system.py

```diff
@@ -439,3 +442,242 @@ def test_normalize(self):
+
+    @pytest.mark.parametrize(
+        'is_representative, has_atom_indices, mol_label_list, n_mol_list, atom_labels_list, composition_formula_list, custom_formulas',
+        [
+            (
+                True,
+                True,
+                ['H20'],
+                [3],
+                [['H', 'O', 'O']],
+                ['group_H20(1)', 'H20(3)', 'H(1)O(2)', 'H(1)O(2)', 'H(1)O(2)'],
+                [None, None, None, None, None]
+            ), # pure system
+            (
+                False,
+                True,
+                ['H20'],
+                [3],
+                [['H', 'O', 'O']],
+                [None, None, None, None, None],
+                [None, None, None, None, None]
+            ), # non-representative system
+            (
+                True,
+                True,
+                [None],
+                [3],
+                [['H', 'O', 'O']],
+                ['Unknown(1)', 'Unknown(3)', 'H(1)O(2)', 'H(1)O(2)', 'H(1)O(2)'],
+                [None, None, None, None, None]
+            ), # missing branch labels
+            (
+                True,
+                True,
+                ['H20'],
+                [3],
+                [[None, None, None]],
+                ['group_H20(1)', 'H20(3)', 'Unknown(3)', 'Unknown(3)', 'Unknown(3)'],
+                [None, None, None, None, None]
+            ), # missing atom labels
+            (
+                True,
+                False,
+                ['H20'],
+                [3],
+                [['H', 'O', 'O']],
+                ['group_H20(1)', 'H20(3)', None, None, None],
+                [None, None, None, None, None]
+            ), # missing atom indices
+            (
+                True,
+                True,
+                ['H20'],
+                [3],
+                [['H', 'O', 'O']],
+                ['waters(1)', 'water_molecules(3)', 'H(1)O(2)', 'H(1)O(2)', 'H(1)O(2)'],
+                ['waters(1)', 'water_molecules(3)', None, None, None]
+            ), # custom formulas
+            (
+                True,
+                True,
+                ['H20', 'Methane'],
+                [5, 2],
+                [['H', 'O', 'O'], ['C', 'H', 'H', 'H', 'H']],
+                ['group_H20(1)group_Methane(1)', 'H20(5)', 'H(1)O(2)', 'H(1)O(2)', 'H(1)O(2)', 'H(1)O(2)', 'H(1)O(2)', 'Methane(2)', 'C(1)H(4)', 'C(1)H(4)'],
+                [None, None, None, None, None, None, None, None, None, None]
+            ), # binary mixture
+        ],
+    )
+    def test_system_hierarchy_for_molecules(
+        self,
+        is_representative: bool,
+        has_atom_indices: bool,
+        mol_label_list: List[str],
+        n_mol_list: List[int],
+        atom_labels_list: List[str],
+        composition_formula_list: List[str],
+        custom_formulas: List[str]
+    ):
+        """
+        Test the `ModelSystem` normalization of 'composition_formula' for atoms and molecules.
+
+        Description of test parameters:
```

Comment:
I was following the idea of not describing the inputs in testing, but it might be a good idea. I suggest then autoDocstring as I said

Reply:
I'm fine if you don't want to have this, I just thought it might be nice to include in this case cause the testing function is a bit complicated

Reply:
Yes, it is very nice. Just the style we can combine it with other functions across the code